server/controllers/finance/debtors/index.js

```diff
@@ -279,21 +279,33 @@ function financialPatient(debtorUuid) {
   let sql = `
     SELECT transaction.trans_id, BUID(transaction.entity_uuid) as entity_uuid, transaction.description,
       BUID(transaction.record_uuid) AS record_uuid, transaction.trans_date,
-      SUM(transaction.credit_equiv) AS credit, SUM(transaction.debit_equiv) AS debit,
-      IF(ISNULL(transaction.reference), '', CONCAT_WS('.', '${identifiers.INVOICE.key}', transaction.abbr, transaction.reference)) AS reference
+      SUM(transaction.credit_equiv) AS credit, SUM(transaction.debit_equiv) AS debit, transaction.reference, transaction.referenceCash,
+      transaction.referenceVoucher 
     FROM (
         SELECT posting_journal.trans_id, posting_journal.entity_uuid, posting_journal.description, posting_journal.record_uuid,
-          posting_journal.trans_date, posting_journal.debit_equiv, posting_journal.credit_equiv, invoice.reference, project.abbr
+          posting_journal.trans_date, posting_journal.debit_equiv, posting_journal.credit_equiv, doc1.text AS reference, project.abbr, doc2.text AS referenceCash,
+          doc3.text AS referenceVoucher
         FROM posting_journal
           LEFT JOIN invoice ON invoice.uuid = posting_journal.record_uuid
-          LEFT JOIN project ON invoice.project_id = project.id
+          LEFT JOIN cash ON cash.uuid = posting_journal.record_uuid
```

Comment:
@lomamech, I appreciate the hard work you've done here and your attention to detail.  However, this is much easier for the database:
```sql
  SELECT BUID(record_uuid) as record_uuid, trans_id, trans_date, account_id, credit_equiv, debit_equiv,
    BUID(reference_uuid) AS reference_uuid, description, BUID(entity_uuid) as entity_uuid, document_map.text
  FROM combined_ledger LEFT JOIN document_map ON combined_ledger.record_uuid = document_map.uuid
  WHERE combined_ledger.entity_uuid = ?
```

Only one JOIN required for the text label.  No need to JOIN on `cash`, `voucher`, and `invoice`.  Also, if we ever add another table in the future, there will be no need to update this SQL because the document will be written to `document_map` with a TRIGGER.

For better performance, you could do the JOIN after the WHERE .. I'm not sure if MySQL would optimize this for you or not.